shared/api/auth.go

```diff
@@ -84,6 +84,11 @@ func (i *Identity) SetWritable(put IdentityPut) {
 	i.TLSCertificate = put.TLSCertificate
 }
 
+// AddEntitlement adds an entitlement to the identity.
+func (i *Identity) AddEntitlement(entitlement string) {
+	i.Entitlements = append(i.Entitlements, entitlement)
+}
+
```

Comment:
I'm not sure but can this function just be defined on the `AddedEntitlements` struct? It's embedded, so the parent struct will then implement the method automatically and you won't need to implement it separately everywhere.